velox/exec/WindowFunction.h

```diff
@@ -129,6 +129,12 @@ class WindowFunction {
   SelectivityVector invalidRows_;
 };
 
+struct WindowFunctionMetadata {
+  /// True if the WindowFunction support streaming calculation. For
+  /// example, rank() and row_number().
+  bool streaming{false};
```

Comment:
This name is not very clear. All window function support streaming execution at the partition granularity. However, some support streaming at row granilarity. Perhaps, ProcessingUnit enum with two values, kPartition and kRow, would work better. rank and row_number window functions support streaming per row. Also, all aggregate functions support that.

```
enum class ProcessingUnit {
   kPartition,
   kRow,
}
```

Reply:
@mbasmanova Great suggestions. I'll make the updates later. @aditi-pandit, I'm considering separating the WindowFunction metadata changes into a distinct PR. However, testing is challenging since the support for `kRow `is only included in the subsequent PR. What do you think about merging the WindowFunction metadata changes and rank optimization into a single PR, and then addressing the aggregate window function with the default frame in a separate PR?